subgraphs/notional-finance/src/setters/financialMetrics.ts

```diff
@@ -26,63 +27,89 @@ import {
 import { getOrCreateLendingProtocol } from "../getters/protocol";
 import { getOrCreateToken } from "../getters/token";
 import { getOrCreateInterestRate } from "../getters/interestRate";
-import { MarketDailySnapshot } from "../../generated/schema";
 
-export function updateFinancials(
-  event: ethereum.Event,
-  marketId: string,
-  prevMarketDailySnapshot: MarketDailySnapshot
-): void {
+export function updateRevenues(event: ethereum.Event, marketId: string): void {
   const market = getOrCreateMarket(event, marketId);
   const protocol = getOrCreateLendingProtocol();
-  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, marketId);
-  const financialsDailySnapshots = getOrCreateFinancialsDailySnapshot(event);
-
   const dailyId = event.block.timestamp.toI64() / SECONDS_PER_DAY;
   const marketDailySnapshot = getOrCreateMarketDailySnapshot(
     event,
     dailyId,
     marketId
   );
+  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, marketId);
+  const financialsDailySnapshots = getOrCreateFinancialsDailySnapshot(event);
 
-  // total/cumulative revenue calculation (at event time)
-  const dailyLendingRate = getOrCreateInterestRate(market.id).rate.div(
-    DAYS_PER_YEAR
+  // daysSincePrevSnapshot
+  let prevDailyId: i64;
+  if (market._prevDailyId == "") {
+    prevDailyId = dailyId;
+  } else {
+    prevDailyId = BigInt.fromString(market._prevDailyId).toI64();
+  }
+  const prevMarketDailySnapshot = getOrCreateMarketDailySnapshot(
+    event,
+    prevDailyId,
+    marketId
   );
-  const dailySupplySideRevenueUSD =
-    prevMarketDailySnapshot.totalDepositBalanceUSD.times(dailyLendingRate);
-  const dailyProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
-  const dailyTotalRevenueUSD = dailySupplySideRevenueUSD.plus(
-    dailyProtocolSideRevenueUSD
+  const daysSincePrevSnapshot = new BigDecimal(
```

Comment:
We should have some sanity checks to see if prev > current. There are negative revenues in the syncing subgraph I see

Reply:
Prev < Current is never possible. Exploring the cause of the issue.

Reply:
The issue is stemming from negative inputTokenBalance / totalDepositBalanceUSD. I've not been able to debug further because graph deploys have been failing for me all day.